src/bcf/mod.rs

```diff
@@ -303,6 +303,158 @@ impl Drop for IndexedReader {
     }
 }
 
+/// This module contains bitmask constants for `SyncedReader`.
+pub mod sr_pairing {
+    /// Allow different alleles, as long as they all are SNPs.
+    pub const SNPS: u32 = ::htslib::BCF_SR_PAIR_SNPS;
+    /// The same as above, but with indels.
+    pub const INDELS: u32 = ::htslib::BCF_SR_PAIR_INDELS;
+    /// Any combination of alleles can be returned by `bcf_sr_next_line()`.
+    pub const ANY: u32 = ::htslib::BCF_SR_PAIR_ANY;
+    /// At least some of multiallelic ALTs must match.  Implied by all the others with the exception of `EXACT`.
+    pub const SOME: u32 = ::htslib::BCF_SR_PAIR_SOME;
+    /// Allow REF-only records with SNPs.
+    pub const SNP_REF: u32 = ::htslib::BCF_SR_PAIR_SNP_REF;
+    /// Allow REF-only records with indels.
+    pub const INDEL_REF: u32 = ::htslib::BCF_SR_PAIR_INDEL_REF;
+    /// Require the exact same set of alleles in all files.
+    pub const EXACT: u32 = ::htslib::BCF_SR_PAIR_EXACT;
+    /// `SNPS | INDELS`.
+    pub const BOTH: u32 = ::htslib::BCF_SR_PAIR_BOTH;
+    /// `SNPS | INDELS | SNP_REF | INDEL_REF`.
+    pub const BOTH_REF: u32 = ::htslib::BCF_SR_PAIR_BOTH_REF;
+}
+
+/// A wrapper for `bcf_srs_t`; allows joint traversal of multiple VCF and/or BCF files.
+#[derive(Debug)]
+pub struct SyncedReader {
+    /// Internal handle for the synced reader.
+    inner: *mut htslib::bcf_srs_t,
+
+    /// RC's of `HeaderView`s of the readers.
+    headers: Vec<Rc<HeaderView>>,
+}
+
+// TODO: add interface for setting threads, ensure that the pool is freed properly
+
+impl SyncedReader {
+    pub fn new() -> Result<Self, AllocationError> {
+        let inner = unsafe { htslib::bcf_sr_init() };
+        if inner.is_null() {
+            Err(AllocationError::Some)
+        } else {
+            Ok(SyncedReader {
+                inner: inner,
+                headers: Vec::new(),
+            })
+        }
+    }
+
+    /// Enable or disable requiring of index
+    pub fn set_require_index(&mut self, do_require: bool) {
+        unsafe {
+            (*self.inner).require_index = if do_require { 1 } else { 0 };
+        }
+    }
+
+    /// Set the given bitmask of values from `sr_pairing` module.
+    pub fn set_pairing(&mut self, bitmask: u32) {
```

Comment:
I think it would be nicer to have an enum for the different pairings and hide this bitmask details inside `set_pairing`.

Reply:
I think about any combination of bitmasks is possible, so I would argue against this. When the dust settles a bit on Rust creates, we might rely on an external crate here, but I would refrain from that until there crystalizes a best practice for bit masks.

Reply:
Ah, sorry, now I get it. Indeed, I agree.

Reply:
👍